src/api/Feed.js

```diff
@@ -184,14 +185,73 @@ export const getParsedFileActivitiesResponse = (response?: { entries: FileActivi
+
+                    versionsItem.type = 'file_version';
+                    if (versionsItem.action_by) {
+                        const collaborators = {};
+
+                        if (versionsItem.action_by.length === 1) {
+                            versionsItem.uploader_display_name = versionsItem.action_by[0].name;
+                        }
+
+                        versionsItem.action_by.map(collaborator => {
+                            collaborators[collaborator.id] = { ...collaborator };
+                            return collaborator;
+                        });
+
+                        versionsItem.collaborators = collaborators;
+                    }
+                    if (versionsItem.end?.number !== undefined) {
+                        versionsItem.version_end = versionsItem.end.number;
+                        versionsItem.id = versionsItem.end.id;
+                    }
+                    if (versionsItem.start?.number !== undefined) {
+                        versionsItem.version_start = versionsItem.start.number;
+                    }
+
+                    if (versionsItem.version_start === versionsItem.version_end) {
+                        versionsItem.version_number = versionsItem.version_start;
+
+                        if (
+                            versionsItem.action_type === 'created' &&
+                            versionsItem.start?.created_at &&
+                            versionsItem.start?.created_by
+                        ) {
+                            versionsItem.modified_at = versionsItem.start.created_at;
+                            versionsItem.modified_by = { ...versionsItem.start.created_by };
+                        }
+                        if (
+                            versionsItem.action_type === 'trashed' &&
+                            versionsItem.start?.trashed_at &&
+                            versionsItem.start?.trashed_by
+                        ) {
+                            versionsItem.trashed_at = versionsItem.start.trashed_at;
+                            versionsItem.trashed_by = { ...versionsItem.start.trashed_by };
+                        }
+                        if (
+                            versionsItem.action_type === 'restored' &&
+                            versionsItem.start?.restored_at &&
+                            versionsItem.start?.restored_by
+                        ) {
+                            versionsItem.restored_at = versionsItem.start.restored_at;
+                            versionsItem.restored_by = { ...versionsItem.start.restored_by };
+                        }
+                    }
+
+                    return versionsItem;
+                }
+
                 default: {
                     return null;
                 }
             }
         })
-        .filter(item => !!item);
+        .filter(item => !!item)
+        .reverse();
 
-    return { entries: parsedData };
+    return parsedData;
```

Comment:
you could move the `return` to the `map` directly instead of needing the variable

Reply:
I prefer having the variable for the typing and debugging tradeoffs